pkg/client/linstor.go

```diff
@@ -1341,3 +1297,71 @@ func (s *Linstor) GetNodeTopologies(ctx context.Context, nodename string) (*csi.
 
 	return topo, nil
 }
+
+// Delete the given resource definition and linked resource group, but only if:
+// * No resources are placed
+// * No snapshots of the resource exist
+func (s *Linstor) deleteResourceDefinitionAndGroupIfUnused(ctx context.Context, rdName string) error {
+	log := s.log.WithField("rd", rdName)
+	log.Info("checking RD for removal")
+
+	rDef, err := s.client.ResourceDefinitions.Get(ctx, rdName)
+	if err != nil {
+		// if no RD was found, returns nil
+		return nil404(err)
+	}
+
+	log.Debug("checking for undeleted resources")
+
+	resources, err := s.client.Resources.GetAll(ctx, rdName)
```

Comment:
if the comment in the err case is correct, then you don't need the part above (`ResourceDefinitions.Get()`). You need the `rDef` later at the end. You could move the `rDef` part down. IMO it would make it a bit easier to read.

Reply:
Done. Almost moved that part to after it was already deleted :sweat_smile: